src/core/server/status/plugins_status.ts

```diff
@@ -57,114 +97,139 @@ export class PluginsStatusService {
+      this.pluginStatus[plugin] = status;
+
+      if (status.level !== previousReportedLevel) {
+        this.updatePluginStatuses$.next(plugin);
+      }
+    });
   }
 
   public blockNewRegistrations() {
     this.newRegistrationsAllowed = false;
   }
 
   public getAll$(): Observable<Record<PluginName, ServiceStatus>> {
-    if (!this.allPluginsStatuses) {
-      this.allPluginsStatuses = this.getPluginStatuses$([...this.deps.pluginDependencies.keys()]);
-    }
-    return this.allPluginsStatuses;
+    return this.pluginStatus$.asObservable();
   }
 
   public getDependenciesStatus$(plugin: PluginName): Observable<Record<PluginName, ServiceStatus>> {
-    const dependencies = this.deps.pluginDependencies.get(plugin);
-    if (!dependencies) {
-      throw new Error(`Unknown plugin: ${plugin}`);
-    }
-    if (!this.dependenciesStatuses.has(plugin)) {
-      this.dependenciesStatuses.set(
-        plugin,
-        this.getPluginStatuses$(dependencies).pipe(
-          // Prevent many emissions at once from dependency status resolution from making this too noisy
-          debounceTime(25)
-        )
-      );
-    }
-    return this.dependenciesStatuses.get(plugin)!;
+    const directDependencies = this.pluginData[plugin].dependencies;
+
+    return this.pluginStatus$.asObservable().pipe(
+      map((allStatus) =>
+        Object.keys(allStatus)
+          .filter((dep) => directDependencies.includes(dep))
+          .reduce((acc: PluginStatus, key: PluginName) => {
+            acc[key] = allStatus[key];
+            return acc;
+          }, {})
+      ),
+      distinctUntilChanged()
+    );
   }
 
   public getDerivedStatus$(plugin: PluginName): Observable<ServiceStatus> {
-    if (!this.derivedStatuses.has(plugin)) {
-      this.derivedStatuses.set(
-        plugin,
-        this.update$.pipe(
-          debounceTime(25), // Avoid calling the plugin's custom status logic for every plugin that depends on it.
-          switchMap(() => {
-            // Only go up the dependency tree if any of this plugin's dependencies have a custom status
-            // Helps eliminate memory overhead of creating thousands of Observables unnecessarily.
-            if (this.anyCustomStatuses(plugin)) {
-              return combineLatest([this.deps.core$, this.getDependenciesStatus$(plugin)]).pipe(
-                map(([coreStatus, pluginStatuses]) => {
-                  return getSummaryStatus(
-                    [...Object.entries(coreStatus), ...Object.entries(pluginStatuses)],
-                    {
-                      allAvailableSummary: `All dependencies are available`,
-                    }
-                  );
-                })
-              );
-            } else {
-              return this.defaultInheritedStatus$;
-            }
-          })
-        )
-      );
-    }
-    return this.derivedStatuses.get(plugin)!;
+    return this.pluginData$.asObservable().pipe(
+      pluck(plugin, 'derivedStatus'),
```

Comment:
optional: someone might be unfamiliar with `pluck` API, I'd use
```js
map(plugins => plugin[name].derivedStatus)
```
 to simplify code